src/metadata/matroska_handler.cc

```diff
@@ -152,6 +152,7 @@ void MatroskaHandler::parseMKV(const std::shared_ptr<CdsItem>& item, MemIOHandle
 {
     file_io_callback ebml_file(item->getLocation().c_str());
     EbmlStream ebml_stream(ebml_file);
+    *p_io_handler = nullptr;
```

Comment:
That has to be moved to `serveContent`. The method is called with `nullptr` from `fillMetadata`.